src/main/java/edu/harvard/iq/dataverse/api/Files.java

```diff
@@ -392,14 +392,26 @@ public Response updateFileMetadata(@FormDataParam("jsonData") String jsonData,
                 javax.json.JsonObject jsonObject = jsonReader.readObject();
                 String label = jsonObject.getString("label", null);
                 String directoryLabel = jsonObject.getString("directoryLabel", null);
+                // If the user is trying to change the label/directoryLabel or not.
+                boolean labelChange = true;
+                if (label == null && directoryLabel == null) {
+                    labelChange = false;
+                }
+                String oldLabel = df.getFileMetadata().getLabel();
+                String oldDirectoryLabel = df.getFileMetadata().getDirectoryLabel();
+                String oldPathPlusName = oldDirectoryLabel + oldLabel;
+                String incomingPathPlusName = directoryLabel + label;
+                if (oldPathPlusName.equals(incomingPathPlusName)) {
+                    labelChange = false;
+                }
                 String path = "";
                 if (directoryLabel != null) {
                     path = directoryLabel + "/";
                 }
                 if (label == null) {
-                    label = df.getFileMetadata().getLabel();
+                    label = oldLabel;
                 }
-                if (IngestUtil.conflictsWithExistingFilenames(label, directoryLabel, fmdList, df)) {
```

Comment:
if directoryLabel is not passed in, do we want path to be ""? or keep the old value (as you do with label)?

Reply:
`path` is only used in the error message...

- "Filename already exists at README.md"
- "Filename already exists at code/README.md"

... and I'm happy with how it works. So yes, I think leaving it as "" (empty string) is fine.

Reply:
Sorry path confused me here, what I'm getting it is:

Why don't we have have a 
`
                if (directoryLabel == null) {
                    directoryLabel = oldDrectorylLabel;
                }
`

like with label, otherwise null gets passed into the directory+name check, no? (and if we did this, we could really get rid of path altogether and just use directoryLabel + /. And this way if directorLabel were null, we would get a more accurate message anyway:

"Filename already exists at code/README.md", when "Filename already exists at README.md" isn't accurate?

Reply:
I got rid of the extra path variable and made sure the messages are accurate.